packages/sysweb3-keyring/src/trezor/index.ts

```diff
@@ -1,2 +1,632 @@
-// export * from './transactions';
-export * from './init';
+/* eslint-disable camelcase */
+import TrezorConnect, {
+  AccountInfo,
+  DEVICE_EVENT,
+  SignTransaction,
+} from '@trezor/connect-web';
+import { Buffer } from 'buffer';
+import { ethers } from 'ethers';
+import Web3 from 'web3';
+
+const initialHDPath = `m/44'/60'/0'/0/0`;
+const pathBase = `m/44'/60'/0'/0`;
+// const MAX_INDEX = 1000;
+const DELAY_BETWEEN_POPUPS = 1000;
+const TREZOR_CONNECT_MANIFEST = {
+  appUrl: 'https://paliwallet.com/',
+  email: 'pali@pollum.io',
+};
+
+export interface TrezorControllerState {
+  hdPath: string;
+  paths: Record<string, number>;
+}
+
+interface ISignUtxoTx extends SignTransaction {
+  coin: string;
+}
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  interface Window {
+    TrezorConnect: any;
+  }
+}
+export class TrezorKeyring {
+  public hdPath: string = initialHDPath;
+  public publicKey: Buffer;
+  public chainCode: Buffer;
+  public paths: Record<string, number> = {};
+  public model?: string;
+
+  constructor() {
+    this.publicKey = Buffer.from('', 'hex');
+    this.chainCode = Buffer.from('', 'hex');
+    this.hdPath = '';
+    this.paths = {};
+    TrezorConnect.on(DEVICE_EVENT, (event: any) => {
+      if (event.payload.features) {
+        this.model = event.payload.features.model;
+      }
+    });
+    this.init();
+  }
+
+  /**
+   * Initialize Trezor script.
+   *
+   * Trezor Connect raises an error that reads "Manifest not set" if manifest is not provided. It can be either set via manifest method or passed as a param in init method.
+   * @returns true, if trezor was initialized successfully and false, if some error happen
+   */
+
+  private async init() {
+    window.TrezorConnect = TrezorConnect;
```

Comment:
its missing a validation to check if window is existent, there might be usage cases in which window is undefined and than we should disable trezor from initialising as a safety measure